### ISSUE TITLE
Backup "repos_filtered.json" and "repos.json" before `all-repos-clone` and restore them on `KeyboardInterrupt`

### ISSUE THREAD

**Issue by simon-liebehenschel** [asottile/all-repos#295]
## TL;DR

Not a bug. This is only a minor improvement request. The end users can use my workaround. If you think that adding 5-6 lines of code does not worth it, I do not mind if this enhancement proposal will be closed.

**I will close this right away to save your time, but if a PR is welcome, then you can write in the comments.**

## Current behavior

1. Run `all-repos-clone`
2. Press `Ctrl + C`.
3. Run `all-repos-clone` again.

Result:
```shell
output_dir should only contain repos.json, repos_filtered.json, and directories
```
Imagine a world where any terminated program also drops its config files to prevent rerun. Not excited, is not it? :smile: 

## Expected behavior

 1. Run `all-repos-clone`
2. Press `Ctrl + C`.
3. Run `all-repos-clone` again. No errors.

## Workaround

If my `all-repos-clone` was interrupted for some reason, then I comment out https://github.com/asottile/all-repos/blob/main/all_repos/config.py#L63 before the next `all-repos-clone` execution. This way I can restore "repos_filtered.json" and "repos.json" files.

## Pseudo-code just to explain the idea

```python
# Backup "repos_filtered.json" and "repos.json" files to a temporary directory.
...

try:
    # Call "all-repos-clone"...
except KeyboardInterrupt:
    # Restore "repos_filtered.json" and "repos.json" files
    ...
    # Quit.
    sys.exit()
```
